pkg/logql/log/label_filter_test.go

```diff
@@ -350,3 +351,32 @@ func TestReduceAndLabelFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestStringLabelFilter(t *testing.T) {
+	tests := []struct {
+		name        string
+		filter      *StringLabelFilter
+		labels      labels.Labels
+		shouldMatch bool
+	}{
+		{
+			name:        "without-label",
+			filter:      NewStringLabelFilter(labels.MustNewMatcher(labels.MatchNotEqual, "subqueries", "0")),
+			labels:      labels.Labels{{Name: "msg", Value: "hello"}}, // no label `subqueries`
+			shouldMatch: false,
+		},
```

Comment:
It would also be nice to add a test where `subqueries` is "0" and another where it is the empty string:
```
labels.Labels{{Name: "msg", Value: "hello"}, {Name: "subqueries", Value: "0"}},
labels.Labels{{Name: "msg", Value: "hello"}, {Name: "subqueries", Value: ""}},
```

Reply:
good point and nice catch. I considered `subqueries !="0"` response should be exactly same as `subqueries != 0`. But I failed to consider the case what happens if `subquerie =""`. You are right it should match lines that doesn't have label for queries like `subqueries=""` and `subqueries~=""`

Fixed it in https://github.com/grafana/loki/pull/6766/commits/bab6ca046c0382702aa0992577a8d9db5dc01342

does that sounds better?

Reply:
Looks good, thanks 👍